docs/en/concepts-and-designs/mal.md

```diff
@@ -0,0 +1,188 @@
+{region="asia-north",az="az-1"} 0.3333  // 11 / 33
+```
+
+### Aggregation Operation
+
+Sample family supports the following aggregation operations that can be used to aggregate the samples of a single sample family,
+resulting in a new sample family of fewer samples(even single one) with aggregated values:
+
+ - sum (calculate sum over dimensions)
+ - min (select minimum over dimensions)
+ - max (select maximum over dimensions)
+ - avg (calculate the average over dimensions)
+ 
+These operations can be used to aggregate over all label dimensions or preserve distinct dimensions by inputting `by` parameter. 
+
+```
+<aggr-op>(by: <tag1, tag2, ...>)
+```
+
+Example expression:
+
+```
+instance_trace_count.sum(by: ['az'])
+```
+
+will output a result:
+
+```
+instance_trace_count{az="az-1"} 133 // 100 + 33
+instance_trace_count{az="az-3"} 20
+```
+
+### Function
+
+`Duraton` is a textual representation of a time range. The formats accepted are based on the ISO-8601 duration format {@code PnDTnHnMn.nS}
+ with days considered to be exactly 24 hours.
+
+Examples:
+ - "PT20.345S" -- parses as "20.345 seconds"
+ - "PT15M"     -- parses as "15 minutes" (where a minute is 60 seconds)
+ - "PT10H"     -- parses as "10 hours" (where an hour is 3600 seconds)
+ - "P2D"       -- parses as "2 days" (where a day is 24 hours or 86400 seconds)
+ - "P2DT3H4M"  -- parses as "2 days, 3 hours and 4 minutes"
+ - "P-6H3M"    -- parses as "-6 hours and +3 minutes"
+ - "-P6H3M"    -- parses as "-6 hours and -3 minutes"
+ - "-P-6H+3M"  -- parses as "+6 hours and -3 minutes"
+
+#### increase
+`increase(Duration)`. Calculates the increase in the time range.
+
+#### rate
+`rate(Duration)`. Calculates the per-second average rate of increase of the time range.
+
+#### irate
+`irate()`. Calculates the per-second instant rate of increase of the time range.
+
+#### tag
+`tag({allTags -> })`. Update tags of samples. User can add, drop, rename and update tags.
+
+#### histogram
+`histogram(le: '<the tag name of le>')`. Transforms less based histogram buckets to meter system histogram buckets. 
+`le` parameter hints the tag name of a bucket. 
+
+#### histogram_quantile
+`histogram_quantile([<φ scalar>])`. Hints meter-system to calculates the φ-quantile (0 ≤ φ ≤ 100) from the buckets. 
+
+## Down Sampling Operation
+MAL should instruct meter-system how to do downsampling for metrics. It doesn't only refer to aggregate raw samples to 
+`minute` level, but also hints data from `minute` to higher levels, for instance, `hour` and `day`. 
+
+Down sampling operations are as global function in MAL:
+
+ - avg
+ - latest
+ - min
+ - max
+ - mean
+ - sum
+ - count
+
+The default one is `avg` if not specific an operation.
+
+If user want get latest time from `last_server_state_sync_time_in_seconds`:
+
+```
+latest(last_server_state_sync_time_in_seconds.tagEqual('production', 'catalog'))
+
+or
+
+latest last_server_state_sync_time_in_seconds.tagEqual('production', 'catalog')
+```
```

Comment:
Could you share a complex expression(including filter, function, and operation) at the end of this document? It would be more clear for the readers.

Reply:
Sure. I prefer to add more examples once most of the features are accomplished though.